tests/backup/backup_helper.go

```diff
@@ -5559,3 +5678,242 @@ func UpgradeKubevirt(versionToUpgrade string, workloadUpgrade bool) error {
+	backupNameSpaces []string) error {
+
+	currentAdminNamespace, _ := getCurrentAdminNamespace()
+	if len(backupNameSpaces) == 1 {
+		currentAdminNamespace = backupNameSpaces[0]
+	}
+	log.Infof("Current CR Namespace: [%s]", currentAdminNamespace)
+
+	backupDriver := Inst().Backup
+	clusterReq := &api.ClusterInspectRequest{OrgId: orgID, Name: clusterName, IncludeSecrets: true, Uid: clusterUID}
+	clusterResp, err := backupDriver.InspectCluster(ctx, clusterReq)
+	if err != nil {
+		return err
+	}
+	clusterObj := clusterResp.GetCluster()
+
+	validateBackupCRInNamespace := func() (interface{}, bool, error) {
+		allBackupCrs, err := ListBackupCRs(currentAdminNamespace, clusterObj)
+		if err != nil {
+			return nil, true, err
+		}
+		log.InfoD("All backup CRs in [%s] are [%v]", currentAdminNamespace, allBackupCrs)
+
+		for _, eachCR := range allBackupCrs {
+			if strings.Contains(eachCR, backupName) {
+				log.Infof("Backup CR found for [%s] under [%s] namespace", backupName, currentAdminNamespace)
+				return nil, false, nil
+			}
+		}
+		return nil, true, fmt.Errorf("Unable to find CR for [%s] under [%s] namespace", backupName, currentAdminNamespace)
+	}
+
+	_, err = task.DoRetryWithTimeout(validateBackupCRInNamespace, 5*time.Minute, 500*time.Millisecond)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Validates Restore CRs created
+func ValidateRestoreCRs(restoreName string, clusterName string, orgID string, clusterUID string, ctx context.Context,
+	restoreNameSpaces map[string]string) error {
+	currentAdminNamespace, _ := getCurrentAdminNamespace()
+	if len(restoreNameSpaces) == 1 {
+		for _, val := range restoreNameSpaces {
+			currentAdminNamespace = val
+		}
+	}
+	log.Infof("Current CR Namespace: [%s]", currentAdminNamespace)
+
+	backupDriver := Inst().Backup
+	clusterReq := &api.ClusterInspectRequest{OrgId: orgID, Name: clusterName, IncludeSecrets: true, Uid: clusterUID}
+	clusterResp, err := backupDriver.InspectCluster(ctx, clusterReq)
+	if err != nil {
+		return err
+	}
+	clusterObj := clusterResp.GetCluster()
+
+	validateRestoreCRInNamespace := func() (interface{}, bool, error) {
+		allRestoreCrs, err := ListRestoreCRs(currentAdminNamespace, clusterObj)
+		if err != nil {
+			return nil, true, err
+		}
+		log.InfoD("All restore CRs in [%s] are [%v]", currentAdminNamespace, allRestoreCrs)
+
+		for _, eachCR := range allRestoreCrs {
+			if strings.Contains(eachCR, restoreName) {
+				log.Infof("Restore CR found for [%s] under [%s] namespace", restoreName, currentAdminNamespace)
+				return nil, false, nil
+			}
+		}
+		return nil, true, fmt.Errorf("Unable to find CR for [%s] under [%s] namespace", restoreName, currentAdminNamespace)
+	}
+
+	_, err = task.DoRetryWithTimeout(validateRestoreCRInNamespace, 5*time.Minute, 500*time.Millisecond)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// ListBackupCRs lists all the Backup CRs present under given namespace
+func ListBackupCRs(
```

Comment:
nit: Can you rename this to `GetBackupCRs`?

Reply:
Done